common/src/services/environment.service.ts

```diff
@@ -2,31 +2,112 @@ import { EnvironmentUrls } from '../models/domain/environmentUrls';
 
-import { ApiService } from '../abstractions/api.service';
-import { EnvironmentService as EnvironmentServiceAbstraction } from '../abstractions/environment.service';
-import { NotificationsService } from '../abstractions/notifications.service';
+import { EnvironmentService as EnvironmentServiceAbstraction, Urls } from '../abstractions/environment.service';
 import { StorageService } from '../abstractions/storage.service';
 
 export class EnvironmentService implements EnvironmentServiceAbstraction {
-    baseUrl: string;
-    webVaultUrl: string;
-    apiUrl: string;
-    identityUrl: string;
-    iconsUrl: string;
-    notificationsUrl: string;
-    eventsUrl: string;
-    enterpriseUrl: string;
-
-    constructor(private apiService: ApiService, private storageService: StorageService,
-        private notificationsService: NotificationsService) { }
-
-    getWebVaultUrl(): string {
+    private baseUrl: string;
+    private webVaultUrl: string;
+    private apiUrl: string;
+    private identityUrl: string;
+    private iconsUrl: string;
+    private notificationsUrl: string;
+    private eventsUrl: string;
+    private enterpriseUrl: string;
+
+    constructor(private storageService: StorageService) { }
+
+    hasBaseUrl() {
+        return this.baseUrl != null;
+    }
+
+    getNotificationsUrl() {
+        if (this.notificationsUrl != null) {
+            return this.notificationsUrl;
+        }
+
+        if (this.baseUrl != null) {
+            return this.baseUrl + '/notifications';
+        }
+
+        return 'https://notifications.bitwarden.com';
+    }
+
+    getEnterpriseUrl() {
+        if (this.enterpriseUrl != null) {
+            return this.enterpriseUrl;
+        }
+
+        if (this.baseUrl != null) {
+            return this.baseUrl + '/portal';
+        }
+
+        return 'https://portal.bitwarden.com';
+    }
+
+    getWebVaultUrl() {
         if (this.webVaultUrl != null) {
             return this.webVaultUrl;
-        } else if (this.baseUrl) {
+        }
+
+        if (this.baseUrl) {
             return this.baseUrl;
         }
-        return null;
+        return 'https://vault.bitwarden.com';
+    }
+
+    getSendUrl() {
+        return this.getWebVaultUrl() === 'https://vault.bitwarden.com'
+            ? 'https://send.bitwarden.com/#'
+            : this.getWebVaultUrl() + '/#/send/';
+    }
+
+    getIconsUrl() {
+        if (this.iconsUrl != null) {
+            return this.iconsUrl;
+        }
+
+        if (this.baseUrl) {
+            return this.baseUrl + '/icons';
+        }
+
+        return 'https://icons.bitwarden.net';
+    }
+
+    getApiUrl() {
+        if (this.apiUrl != null) {
+            return this.apiUrl;
+        }
+
+        if (this.baseUrl) {
+            return this.baseUrl + '/api';
+        }
+
+        return 'https://api.bitwarden.com';
+    }
+
+    getIdentityUrl() {
+        if (this.identityUrl != null) {
+            return this.apiUrl;
```

Comment:
Shouldn't this return `this.identityUrl` here? I may be missing what's supposed to happen here if `this.identityUrl` is explicitly set.